components/resource_control/src/service.rs

```diff
@@ -139,6 +155,163 @@ impl ResourceManagerService {
             }
         }
     }
+
+    async fn load_controller_config(&self) -> Result<RequestUnitConfig, ()> {
+        loop {
+            match self
+                .pd_client
+                .load_global_config(RESOURCE_CONTROL_CONTROLLER_CONFIG_PATH.to_string())
+                .await
+            {
+                Ok((items, _)) => {
+                    match serde_json::from_slice::<ControllerConfig>(items[0].get_payload()) {
+                        Ok(c) => return Ok(c.request_unit),
+                        Err(e) => {
+                            error!("parse controller config failed"; "err" => ?e);
+                            let _ = GLOBAL_TIMER_HANDLE
+                                .delay(std::time::Instant::now() + RETRY_INTERVAL)
+                                .compat()
+                                .await;
+                            continue;
+                        }
+                    }
+                }
+                Err(err) => {
+                    error!("failed to controller config"; "err" => ?err);
+                    let _ = GLOBAL_TIMER_HANDLE
+                        .delay(std::time::Instant::now() + RETRY_INTERVAL)
+                        .compat()
+                        .await;
+                    continue;
+                }
+            }
+        }
+    }
+
+    // upload ru metrics periodically.
+    pub async fn upload_ru_metrics(&self) {
+        let mut last_consumption_map: HashMap<String, Consumption> = HashMap::new();
+        // load controller config firstly.
+        let config = self.load_controller_config().await.unwrap_or_default();
+
+        loop {
+            let background_groups: Vec<_> = self
+                .manager
+                .resource_groups
+                .iter()
+                .filter_map(|kv| {
+                    let g = kv.value();
+                    g.limiter.as_ref().map(|limiter| GroupStats {
```

Comment:
Why reuse the GroupStats struct, you only depends on the GroupStatistics field here, better define a separated struct here.

Reply:
Yes! fixed it as `UploadStatistic`, PTAL, thx!